qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py

```diff
@@ -74,11 +76,125 @@ def get_raw(self) -> np.ndarray:
                                                 self.steps,
                                                 self.mode)
         else:
-            raise RuntimeError("No instrument attached to Parameter")
+            raise RuntimeError("No instrument attached to Parameter.")
 
         return data
 
 
+class TimeTrace(ParameterWithSetpoints):
+    """
+    A parameter class that holds the data corresponding to the time dependence of
+    current and voltage.
+    """
+
+    def __init__(self, name: str, instrument: Instrument, **kwargs) -> None:
+
+        super().__init__(name=name,
+                         instrument=instrument,
+                         **kwargs)
+
+    def _check_time_trace(self) -> None:
+        """
+        A helper function that compares the integration time with measurement
+        interval for accurate results.
+
+        Raises:
+            RuntimeError: If no instrument attached to Parameter.
+        """
+        if self.instrument is None:
+            raise RuntimeError("No instrument attached to Parameter.")
+
+        dt = self.instrument.dt()
+        nplc = self.instrument.nplc()
+        plc = 1/float(self.instrument.ask('localnode.linefreq'))
+        if nplc * plc > dt:
+            warnings.warn(f'Integration time of {nplc*plc*1000:.1f} ' +
+                          f'ms is longer than {dt*1000:.1f} ms set ' +
+                          'as measurement interval. Consider lowering ' +
+                          'NPLC or increasing interval.', UserWarning, 2)
+
+    def _set_mode(self, mode: str) -> None:
+        """
+        A helper function to set correct units and labels.
+
+        Args:
+            mode: User defined mode for the timetrace. It can be either
+            current, "i", or voltage "v".
+        """
+        if mode == 'i':
+            self.unit='A'
+            self.label='Current'
+        if mode == 'v':
+            self.unit='V'
+            self.label='Voltage'
+
+    def _time_trace(self, npts: int, dt: float, mode: str) -> np.ndarray:
```

Comment:
Why does this method take arguments? All of `npts`, `dt`, and `mode` are attributes of `self`, aren't they?